src/material/badge/badge.ts

```diff
@@ -156,6 +156,20 @@ export class MatBadge extends _MatBadgeBase implements OnInit, OnDestroy, CanDis
       if (nativeElement.nodeType !== nativeElement.ELEMENT_NODE) {
         throw Error('matBadge must be attached to an element node.');
       }
+
+      const badgeTagName: string = 'mat-icon';
+
```

Comment:
I think that we should wrap this warning in `ngDevMode` so it gets stripped from production builds.

Reply:
it is wrapped in `ngDevMode` condition before the `matBadge must be attached to an element node.`, at line 154.

![image](https://github.com/angular/components/assets/54370141/fac31e05-6238-46df-abd8-a3757bffa5cf)